### DISCUSSION TITLE
v2.14.0

### DISCUSSION THREAD

**Discussion by Francessco121** [milestonetg/msal-js-dart#29] [Announcements]
- **(breaking change)** Support for MSAL.js 2.14.
  - MSAL 2.x has a slightly different API than MSAL 1.x, some notable changes include:
    - `UserAgentApplication` was renamed to `PublicClientApplication`.
    - `handleRedirectFuture` replaces `handleRedirectCallback`.
    - Request/response classes have new names.
- **(breaking change)** Removed deprecated `MissingMsalJsException` class.
- Fixed issue where setting the base `Logger` log level would set the wrong level.
- Fixed issue where getting the log level from `LoggerOptions` would return the wrong level.

<hr /><em>This discussion was created from the release <a href='https://github.com/Francessco121/msal-js-dart/releases/tag/v2.14.0'>v2.14.0</a>.</em>

**Comment by mick-stokes** [comment]
Just wondering if you can explain how to log out of a particular account. 
I always get prompted for the account I wish to log out of.

I have tried the following 
    EndSessionPopupRequest? logoutRequest;
     logoutRequest!.account = publicClientApp.getActiveAccount();
     await publicClientApp.logoutPopup(logoutRequest);

But I get an error.

**Reply by Francessco121** [reply]
In the code snippet you provided, `logoutRequest` is always null. I'm not sure what error you are getting but I'm assuming it's that. You need to instantiate a new `EndSessionPopupRequest` before you can set the account.

If that doesn't work, I'd suggest asking for help over at https://github.com/AzureAD/microsoft-authentication-library-for-js (try to provide them a JS snippet since they won't know Dart). This library is just a thin wrapper over the official JS library, unfortunately I can really only help with issues specific to the Dart wrapper itself.
